### ISSUE TITLE
native-codec sample is buggy

### ISSUE THREAD

**Issue by proppy** [googlesamples/android-ndk#40]
it crashes when pressing 'start' then 'rewind':
```
D/HeadsetStateMachine(22751): Disconnected process message: 10, size: 0
V/NativeCodec(32382): @@@ setsurface 0xb497ff08
V/NativeCodec(32382): @@@ create
V/NativeCodec(32382): opening /sdcard/testfile.mp4
V/NativeCodec(32382): failed: -1 (No such file or directory)
D/audio_hw_primary(  357): out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
V/NativeCodec(32382): @@@ rewind
--------- beginning of crash
F/libc    (32382): Fatal signal 11 (SIGSEGV), code 1, fault addr 0xc in tid 32382 (ple.nativecodec)
D/audio_hw_primary(  357): out_set_parameters: enter: usecase(1: low-latency-playback) kvpairs: routing=2
I/DEBUG   (  354): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
I/DEBUG   (  354): Build fingerprint: 'google/shamu/shamu:5.1.1/LYZ28H/1991770:user/release-keys'
I/DEBUG   (  354): Revision: '33696'
I/DEBUG   (  354): ABI: 'arm'
I/DEBUG   (  354): pid: 32382, tid: 32382, name: ple.nativecodec  >>> com.example.nativecodec <<<
I/DEBUG   (  354): signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xc
I/DEBUG   (  354):     r0 0000000c  r1 a1b71df0  r2 00000000  r3 0000000c
I/DEBUG   (  354):     r4 0000000c  r5 12ce8d00  r6 12ce8e40  r7 bea1e1f8
I/DEBUG   (  354):     r8 12ce8e40  r9 b4827800  sl 00000000  fp 00000000
I/DEBUG   (  354):     ip b3683f90  sp bea1e1e8  lr b367ca01  pc b6e7e7ba  cpsr 000f0030
I/DEBUG   (  354): 
I/DEBUG   (  354): backtrace:
I/DEBUG   (  354):     #00 pc 000527ba  /system/lib/libc.so (sem_wait+19)
I/DEBUG   (  354):     #01 pc 000049fd  /data/app/com.example.nativecodec-1/lib/arm/libnative-codec-jni.so (looper::addmsg(loopermessage*, bool)+20)
I/DEBUG   (  354):     #02 pc 000049dd  /data/app/com.example.nativecodec-1/lib/arm/libnative-codec-jni.so (looper::post(int, void*, bool)+72)
I/DEBUG   (  354):     #03 pc 00004897  /data/app/com.example.nativecodec-1/lib/arm/libnative-codec-jni.so (Java_com_example_nativecodec_NativeCodec_rewindStreamingMediaPlayer+50)
I/DEBUG   (  354):     #04 pc 000028f5  /data/dalvik-cache/arm/data@app@com.example.nativecodec-1@base.apk@classes.dex
W/ActivityManager(  816):   Force finishing activity 1 com.example.nativecodec/.NativeCodec
I/DEBUG   (  354): 
I/DEBUG   (  354): Tombstone written to: /data/tombstones/tombstone_02
I/BootReceiver(  816): Copying /data/tombstones/tombstone_02 to DropBox (SYSTEM_TOMBSTONE)
W/InputDispatcher(  816): channel '1a73a803 com.example.nativecodec/com.example.nativecodec.NativeCodec (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
E/InputDispatcher(  816): channel '1a73a803 com.example.nativecodec/com.example.nativecodec.NativeCodec (server)' ~ Channel is unrecoverably broken and will be disposed!
I/Zygote  (  370): Process 32382 exited due to signal (11)
I/WindowState(  816): WIN DEATH: Window{1a73a803 u0 com.example.nativecodec/com.example.nativecodec.NativeCodec}
W/InputDispatcher(  816): Attempted to unregister already unregistered input channel '1a73a803 com.example.nativecodec/com.example.nativecodec.NativeCodec (server)'
I/WindowState(  816): WIN DEATH: Window{1843cdb9 u0 SurfaceView}
W/ActivityManager(  816): Exception thrown during pause
W/ActivityManager(  816): android.os.DeadObjectException
W/ActivityManager(  816): 	at android.os.BinderProxy.transactNative(Native Method)
W/ActivityManager(  816): 	at android.os.BinderProxy.transact(Binder.java:496)
W/ActivityManager(  816): 	at android.app.ApplicationThreadProxy.schedulePauseActivity(ApplicationThreadNative.java:704)
W/ActivityManager(  816): 	at com.android.server.am.ActivityStack.startPausingLocked(ActivityStack.java:825)
W/ActivityManager(  816): 	at com.android.server.am.ActivityStack.finishActivityLocked(ActivityStack.java:2726)
W/ActivityManager(  816): 	at com.android.server.am.ActivityStack.finishTopRunningActivityLocked(ActivityStack.java:2583)
W/ActivityManager(  816): 	at com.android.server.am.ActivityStackSupervisor.finishTopRunningActivityLocked(ActivityStackSupervisor.java:2497)
W/ActivityManager(  816): 	at com.android.server.am.ActivityManagerService.handleAppCrashLocked(ActivityManagerService.java:11509)
W/ActivityManager(  816): 	at com.android.server.am.ActivityManagerService.makeAppCrashingLocked(ActivityManagerService.java:11406)
W/ActivityManager(  816): 	at com.android.server.am.ActivityManagerService.crashApplication(ActivityManagerService.java:12090)
W/ActivityManager(  816): 	at com.android.server.am.ActivityManagerService.handleApplicationCrashInner(ActivityManagerService.java:11601)
W/ActivityManager(  816): 	at com.android.server.am.NativeCrashListener$NativeCrashReporter.run(NativeCrashListener.java:86)
I/WindowState(  816): WIN DEATH: Window{327ec15f u0 SurfaceView}
E/lowmemorykiller(  255): Error opening /proc/32382/oom_score_adj; errno=2
I/ActivityManager(  816): Process com.example.nativecodec (pid 32382) has died
```

**Comment by ggfan** [comment]
might be similar to #41, need copy testfile.mp4  to /sdcard/testfile.mp4

**Comment by proppy** [comment]
Fixed by #118
